client/components/LearnerViewDirective.js

```diff
@@ -477,10 +477,22 @@ tie.directive('learnerView', [function() {
             feedbackDiv.scrollTop = feedbackDiv.scrollHeight +
               additionalHeightForLoadingIndicator;
             $timeout(function() {
-              SolutionHandlerService.processSolutionAsync(
-                tasks[currentTaskIndex], code,
+              var index = 0;
```

Comment:
Do you mean we have to run the tests of different tasks simultaneously? What do you mean by assembling?

Now, with sequential execution, the delay is rather subtle, since existing test cases are simple, plus the fact that total runtime is just the sum of the time for running test cases of first task to current.